### PR TITLE
(feat/stdlib): textReader is functioning as expected

### PR DESCRIPTION
**Pull request by aanthony1243**
### Done checklist
- [ ] docs/SPEC.md updated
- [ ] Test cases written

### CODE REVIEW

**Review by wolffcm — CHANGES_REQUESTED** [review]
Interesting, I could see this being useful! I had some comments and questions.

For a new function it would be good to add a Flux test for it, here's an example:
https://github.com/influxdata/flux/blob/master/stdlib/universe/window_test.flux

File: stdlib/experimental/textreader/textreader.flux [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// Package array provides functions for building tables from Flux arrays.
+//
+// introduced: 0.103.0
+// tags: text,tables
```
Comment (sanderson) [inline stdlib/experimental/textreader/textreader.flux:4]:
```suggestion
// tags: text,tables,inputs
```

File: stdlib/experimental/textreader/textreader.flux [hunk 2]
```diff
@@ -0,0 +1,33 @@
+// Package array provides functions for building tables from Flux arrays.
```
Comment (sanderson) [inline stdlib/experimental/textreader/textreader.flux:1]:
```suggestion
// Package textreader provides functions for building tables from text.
```

File: stdlib/experimental/textreader/textreader.flux [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+//
+// introduced: 0.103.0
+// tags: text,tables
+package textreader
+
+
+// from constructs a table from an arbitrary sample of text
```
Comment (sanderson) [inline stdlib/experimental/textreader/textreader.flux:8]:
```suggestion
// from constructs a table from an arbitrary sample of text.
```

Is this "arbitrary"? From the example below, does the text need to be structured in a specific way? Columns defined by pipes? Could one use (or define) another separator?

File: stdlib/experimental/textreader/textreader.flux [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+//
+// Each line of text is converted into an output row or record.
+// All lines must have the same dimensionality/types
+//
+// ## Parameters
+// - txt: text to process
+// - header: array of strings for column headers.  Default is an empty array which indicates using the first
+//           of text as the header
```
Comment (sanderson) [inline stdlib/experimental/textreader/textreader.flux:16]:
```suggestion
// - txt: Text to process.
// - header: List of column headers. Default is `[]`.
//  
//     When set an empty array, the function uses the first line of text as
//     column headers.
```

File: stdlib/experimental/textreader/textreader.flux [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+//
+// introduced: 0.103.0
+// tags: text,tables
+package textreader
+
+
+// from constructs a table from an arbitrary sample of text
+//
+// Each line of text is converted into an output row or record.
+// All lines must have the same dimensionality/types
```
Comment (sanderson) [inline stdlib/experimental/textreader/textreader.flux:11]:
```suggestion
// All lines must have the same structure and data types.
```

File: stdlib/experimental/textreader/textreader.flux [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+//
+// ## Parameters
+// - txt: text to process
+// - header: array of strings for column headers.  Default is an empty array which indicates using the first
+//           of text as the header
+//
+// ## Examples
+//
+// ### Build an arbitrary table
```
Comment (sanderson) [inline stdlib/experimental/textreader/textreader.flux:20]:
```suggestion
// ### Build a table from text
```

File: stdlib/experimental/textreader/textreader.flux [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+// ```
+// import "array"
+//
+// text = "
+// a|b|c
+// 1|2|3
+// "
+//
+// > textreader.from(txt: text, header: [)
+// outputs table with single record {a:1, b:2, c:3}
```
Comment (sanderson) [inline stdlib/experimental/textreader/textreader.flux:30]: With a `>` at the front of the line, we don't have to show or describe the output in the example. We'll render the output table in the documentation.

```suggestion
// > textreader.from(txt: text, header: [])
```

**Review by sanderson — CHANGES_REQUESTED** [review]

**Pull request closed without merging**